merlin/models/tf/models/base.py

```diff
@@ -1343,6 +1347,9 @@ def fit(
         x = _maybe_convert_merlin_dataset(x, batch_size, **kwargs)
         self._maybe_set_schema(x)
 
+        if hasattr(x, "batch_size"):
+            self._batch_size = x.batch_size
```

Comment:
We need to save the batch size used during fit so that when we reload the mode we pass it the same shape inputs. There were some tests of the two tower model that required this for example. This might indicate something to further investigate. I'm not sure it should matter what batch size we use when re-loading the model.